soroban-sdk-macros/src/lib.rs

```diff
@@ -95,6 +97,45 @@ pub fn contractimpl(_metadata: TokenStream, input: TokenStream) -> TokenStream {
     }
 }
 
+#[derive(Debug, FromMeta)]
+struct MetadataArgs {
+    key: String,
+    val: String,
+}
+
+#[proc_macro]
+pub fn metadata(metadata: TokenStream) -> TokenStream {
+    let args = parse_macro_input!(metadata as AttributeArgs);
+    let args = match MetadataArgs::from_list(&args) {
+        Ok(v) => v,
+        Err(e) => return e.write_errors().into(),
+    };
+
+    let gen = {
+        //TODO: is key necessary?
+        let meta_v0 = ScMetaV0 {
+            key: args.key.clone().try_into().unwrap(),
+            val: args.val.try_into().unwrap(),
+        };
+        let meta_entry = ScMetaEntry::ScMetaV0(meta_v0);
+
+        let metadata_xdr = meta_entry.to_xdr().unwrap();
+        let metadata_xdr_lit = proc_macro2::Literal::byte_string(metadata_xdr.as_slice());
+        let metadata_xdr_len = metadata_xdr.len();
+
+        // TODO: Handle collisions on "metadata"?
```

Comment:
Good catch, as it is this macro can only be used once in each scope. We'll need to find a way to make either the name of the var unique, or anonymous, or the name of a mod it can live in unique or anonymous.

Reply:
I changed the name to the key.